modules/platforms/cpp/ignite/client/detail/utils.cpp

```diff
@@ -201,19 +201,45 @@ std::vector<std::byte> pack_tuple(
             builder.claim_null();
     }
 
+    std::int32_t written = 0;
     builder.layout();
     for (std::int32_t i = 0; i < count; ++i) {
         const auto &col = sch.columns[i];
         auto col_idx = tuple.column_ordinal(col.name);
 
-        if (col_idx >= 0)
+        if (col_idx >= 0) {
             append_column(builder, col.type, tuple.get(col_idx), col.scale);
-        else {
+            ++written;
+        } else {
             builder.append_null();
             no_value.set(std::size_t(i));
         }
     }
 
+    if (!key_only && written < tuple.column_count()) {
+        std::vector<bool> written_ind(tuple.column_count(), false);
+        for (std::int32_t i = 0; i < count; ++i) {
+            const auto &col = sch.columns[i];
+            auto col_idx = tuple.column_ordinal(col.name);
+
+            if (col_idx >= 0)
+                written_ind[col_idx] = true;
+        }
+
+        std::stringstream unmapped_columns;
+        for (std::int32_t i = 0; i < tuple.column_count(); ++i) {
+            if (written_ind[i])
+                continue;
+            auto &name = tuple.column_name(i);
+            unmapped_columns << name << ",";
+        }
+        auto unmapped_columns_str = unmapped_columns.str();
+        unmapped_columns_str.pop_back();
+
+        assert(!unmapped_columns_str.empty());
+        throw ignite_error("Tuple contains columns that are not present in the table: " + unmapped_columns_str);
```

Comment:
I think we should use the error message format from `TupleMarshallerImpl`: "Key tuple doesn't match schema: schemaVersion=%s, extraColumns=%s"

Reply:
Good point.